examples/bullet_prespawn/src/main.rs

```diff
@@ -30,109 +32,101 @@ use lightyear::connection::netcode::{ClientId, Key};
-        /// If true, disable any rendering-related plugins
-        #[arg(long, default_value = "false")]
-        headless: bool,
-
-        /// If true, enable bevy_inspector_egui
-        #[arg(short, long, default_value = "false")]
-        inspector: bool,
-
-        /// The port to listen on
-        #[arg(short, long, default_value_t = SERVER_PORT)]
-        port: u16,
-
-        /// Which transport to use
-        #[arg(short, long, value_enum, default_value_t = Transports::WebTransport)]
-        transport: Transports,
-
-        /// If true, clients will predict everything (themselves, the ball, other clients)
-        #[arg(long, default_value = "false")]
-        predict: bool,
-    },
-    Client {
-        /// If true, enable bevy_inspector_egui
-        #[arg(short, long, default_value = "false")]
-        inspector: bool,
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum ServerTransports {
+    Udp { local_port: u16 },
+    WebTransport { local_port: u16 },
+    WebSocket { local_port: u16 },
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ServerSettings {
+    /// If true, disable any rendering-related plugins
+    headless: bool,
+
+    /// If true, enable bevy_inspector_egui
+    inspector: bool,
+
+    /// Which transport to use
+    transport: Vec<ServerTransports>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ClientSettings {
+    /// If true, enable bevy_inspector_egui
+    inspector: bool,
+
+    /// The client id
+    client_id: u64,
+
+    /// The client port to listen on
+    client_port: u16,
 
-        #[arg(short, long, default_value_t = 0)]
-        client_id: u64,
+    /// The ip address of the server
+    server_addr: Ipv4Addr,
 
-        /// The port to listen on
-        #[arg(long, default_value_t = CLIENT_PORT)]
-        client_port: u16,
+    /// The port of the server
+    server_port: u16,
 
-        #[arg(long, default_value_t = Ipv4Addr::LOCALHOST)]
-        server_addr: Ipv4Addr,
+    /// Which transport to use
+    transport: ClientTransports,
+}
 
-        #[arg(short, long, default_value_t = SERVER_PORT)]
-        server_port: u16,
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Settings {
+    pub server: ServerSettings,
+    pub client: ClientSettings,
+}
 
-        /// Which transport to use
-        #[arg(short, long, value_enum, default_value_t = Transports::WebTransport)]
-        transport: Transports,
+#[derive(Parser, PartialEq, Debug)]
+enum Cli {
+    #[cfg(not(target_family = "wasm"))]
+    Server,
+    Client {
+        #[arg(short, long, default_value = None)]
+        client_id: Option<u64>,
     },
 }
 
-cfg_if::cfg_if! {
-    if #[cfg(target_family = "wasm")] {
-        fn main() {
-            // NOTE: clap argument parsing does not work on WASM
+fn main() {
+    cfg_if::cfg_if! {
+        if #[cfg(target_family = "wasm")] {
+            std::panic::set_hook(Box::new(console_error_panic_hook::hook));
```

Comment:
doesnt bevy automatically add the console error panic hook?

Reply:
Yes but i was getting errors before bevy was loading (while parsing the settings). I can probably remove it